core/src/main/java/org/jruby/java/codegen/RealClassGenerator.java

```diff
@@ -144,7 +144,7 @@ public static Class createRealImplClass(Class superClass, Class[] interfaces, Ru
 
         Class newClass = defineRealImplClass(ruby, name, superClass, superTypeNames, simpleToAll);
         if (!newClass.isAssignableFrom(interfaces[0])) {
-            new RuntimeException(newClass.getInterfaces()[0].getClassLoader() + " " + interfaces[0].getClassLoader());
+            throw new RuntimeException(newClass.getInterfaces()[0].getClassLoader() + " " + interfaces[0].getClassLoader());
```

Comment:
This should never happen anyway, but it's a good change.

Reply:
Looks like there a number of failures on the Travis build that are triggered by this change: https://travis-ci.org/jruby/jruby/builds/305072247

Haven't dug into this, but I can revert this change if there are too many failing tests to deal with.

Reply:
Still waiting on this. Can revert the change to not shake up the CI, or leave it in :)

Reply:
Revert this change and file a bug about this useless line.

Reply:
#4880 filed